### DISCUSSION TITLE
支持简单 submit 跳转

### DISCUSSION THREAD

**Discussion by bymoye** [PaperStrike/Pjax#69] [Ideas]
我该如何绑定submit事件呢 
暂时没有翻到相关资料 
例如搜索 我想实现Enter后无刷新搜索
但是无从下手orz
目前想到的方法只能通过拼接链接后通过
`this.pjax.loadURL(link).catch(() => {});`
的方式进行

**Comment by PaperStrike** [comment]
目前默认不支持 `<form>` submit 这些，你确实需要自己看情况 `preventDefault`（阻止跳转），拼接得到 URL，再传递给 Pjax 这样子。

`<form>` submit 的确会跳转，而除跳转之外的 submit 流程完整地模拟起来比较繁琐，而且不属于 Pjax 核心，目前 Pjax 就不支持，需要自行处理后再传给 Pjax。未来有一定可能会内置一个精简过的 submit 处理。

**Reply by bymoye** [reply]
其实我指的是单纯的跳转 例如 https://demo.com/?s=搜索 跳转到 https://demo.com/?s=test 类似这样的 看来只能通过拼接传递了

**Reply by PaperStrike** [reply]
@bymoye 判断跳转类型的逻辑就挺繁琐的。`<form>` 上的 `enctype`、`target`、`method`、`action`，就分别会影响跳转网址拼接方式、是否同窗跳转、是否简单跳转、是否同域跳转，这些属性的实际取值还可能被触发跳转的 `<button>`、`<input>` 等元素通过 `formenctype`、`formtarget` 等对应属性替代。为了不影响非简单 submit，Pjax 只能将上述属性都判断起来。而具体的网站可能不需要这样的判断。

---

#70 现在基本实现了简单 submit 的判断和处理，~这两天~在 0.14.0 发布。
